src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs

```diff
@@ -141,10 +141,8 @@ internal int SerializeLogRecord(LogRecord logRecord)
         // `LogRecord.State` and `LogRecord.StateValues` were marked Obsolete in https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334
 #pragma warning disable 0618
         IReadOnlyList<KeyValuePair<string, object>> listKvp;
-        if (logRecord.State == null)
+        if (logRecord.StateValues != null)
```

Comment:
Any particular reason for this change?

I'm getting a bit confused with the comment made in the main repo PR: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4619#discussion_r1244074186

> Since State would have a value I decided Attributes / StateValues should as well.

If that's going to be the case, then maybe we should have the check as `if (logRecord.StateValues.Count >= 0)`?

Reply:
> Any particular reason for this change?

Two reasons...

* Perf improvement. SDK has already done the type checking of `state` as `IReadOnlyList<KeyValuePair<string, object>>` and stored the result in `StateValues` / `Attributes`. So tweaking the order here just optimizes for the common case and saves rework for GenevaExporter.

* Captures more types of `TState`. SDK will also set `StateValues` / `Attributes` for `IEnumerable<KeyValuePair<string, object>>` so now GenevaExporter will get those values too where previously they were dropped.

> If that's going to be the case, then maybe we should have the check as if (logRecord.StateValues.Count >= 0)? Or does this not apply for GenevaExporter?

We could make the logic that but if `StateValues` / `Attributes` is an empty list then `state as IReadOnlyList<KeyValuePair<string, object>>` will yield the same empty list so it is only going to cost perf for the state fallback/retry.